### PR TITLE
get default test option

### PR DESCRIPTION
**Pull request by alanrenmsft**
the same test option setup code has been duplicated in pretty much every extension that has extension unit test suite, moving them to this package to avoid duplication and make it easier to modify. https://github.com/microsoft/azuredatastudio/blob/main/extensions/dacpac/src/test/index.ts#L11

### CODE REVIEW

File: lib/testOptions.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * ADS_TEST_GREP: custom test filter.
+ * ADS_TEST_INVERT_GREP: whether to invert the grep results.
+ * RUN_UNSTABLE_TESTS: whether to run the unstable tests. By default, only the stable tests will be executed unless this variable is set to true. If ADS_TEST_GREP is specified, this variable will be ignored.
+ * ADS_TEST_TIMEOUT: test timeout.
+ * ADS_TEST_RETRIES: number of retries.
+ * BUILD_ARTIFACTSTAGINGDIRECTORY: target directory for the test reports.
+ * @param testSuiteName The test suite name.
+ */
+export function getDefaultMochaOptions(testSuiteName: string): any {
+	const mochaOptions: any = {
```
Comment (Charles-Gagnon) [inline lib/testOptions.ts]: It'd be good to have actual types here - you can just use `@types/mocha` to get those without bringing in the entire library.

File: lib/testOptions.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		mochaOptions.reporter = 'mocha-multi-reporters';
+		mochaOptions.reporterOptions = {
+			reporterEnabled: 'spec, mocha-junit-reporter',
+			mochaJunitReporterReporterOptions: {
+				testsuitesTitle: `${testSuiteName} ${process.platform}`,
+				mochaFile: path.join(artifactDirectory, `test-results/${process.platform}-${testSuiteName.toLowerCase().replace(/[^\w]/g, '-')}-results.xml`)
+			}
+		};
+	}
+	console.log("Test options:" + mochaOptions);
```
Comment (Charles-Gagnon) [inline lib/testOptions.ts]: Isn't this just going to print out `Test options: [Object object]`?

File: lib/testOptions.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+
+/**
+ * Get the default mocha options for ads extension unit test. The following environment variables will be examined.
```
Comment (Charles-Gagnon) [inline lib/testOptions.ts]: `Get default mocha options for an ADS test, using the below environment variables to control what values various options are set to`

File: lib/testOptions.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ * ADS_TEST_GREP: custom test filter.
+ * ADS_TEST_INVERT_GREP: whether to invert the grep results.
+ * RUN_UNSTABLE_TESTS: whether to run the unstable tests. By default, only the stable tests will be executed unless this variable is set to true. If ADS_TEST_GREP is specified, this variable will be ignored.
```
Comment (Charles-Gagnon) [inline lib/testOptions.ts]: `whether to run unstable tests (tests that have @UNSTABLE@ in their name) ...`

File: lib/testOptions.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	if (grep) {
+		mochaOptions.grep = grep;
+		mochaOptions.invert = invert;
+	} else {
+		if (!runUnstableTest) {
+			mochaOptions.grep = "@UNSTABLE@";
+			mochaOptions.invert = true;
+		}
+	}
+	if (testTimeout) {
```
Comment (Charles-Gagnon) [inline lib/testOptions.ts]: `if (!isNaN(testTimeout)) {`

File: lib/testOptions.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+		mochaOptions.timeout = testTimeout;
+	}
+	if (retries) {
+		mochaOptions.retries = retries;
+	}
+
+	if (artifactDirectory) {
+		mochaOptions.reporter = 'mocha-multi-reporters';
```
Comment (Charles-Gagnon) [inline lib/testOptions.ts:43]: You should mention in the description for this function that it requires having the `mocha-multi-reporters` and `mocha-junit-reporter` packages installed as well. 

(and consider adding those as parameters with these default values)

File: lib/testOptions.ts (hunk 1 above)
Comment (alanrenmsft) [reply lib/testOptions.ts]: done, found the typings version that has the invert option defined in the options.

File: lib/testOptions.ts (hunk 2 above)
Comment (alanrenmsft) [reply lib/testOptions.ts]: fixed.

File: lib/testOptions.ts (hunk 3 above)
Comment (alanrenmsft) [reply lib/testOptions.ts]: done

File: lib/testOptions.ts (hunk 4 above)
Comment (alanrenmsft) [reply lib/testOptions.ts]: done

File: lib/testOptions.ts (hunk 6 above)
Comment (alanrenmsft) [reply lib/testOptions.ts:43]: I think we should include those 2 packages to this repo, thoughts?

Comment (Charles-Gagnon) [reply lib/testOptions.ts:43]: Sure. The VS Code one purposely isn't meant to be tied to a specific test runner but I doubt anyone else will be using this currently so might as well make it easier for us. We can always change it later if it turns out other people are interested (that don't want to use mocha)

File: lib/testOptions.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+				mochaFile: path.join(artifactDirectory, `test-results/${process.platform}-${testSuiteName.toLowerCase().replace(/[^\w]/g, '-')}-results.xml`)
+			}
+		};
+	}
+
+	mochaOptions.timeout = testTimeout;
+	mochaOptions.retries = retries;
+
+	console.log('Mocha Options:');
+	console.log(`grep: ${mochaOptions.grep}`);
```
Comment (Charles-Gagnon) [inline lib/testOptions.ts]: You could also just do `JSON.stringify(mochaOptions)`

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**